GVRf/Framework/src/org/gearvrf/FutureWrapper.java

```diff
@@ -0,0 +1,50 @@
+package org.gearvrf;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * This is a wrapper to convert T to Future<T> so that we do not need many
+ * versions of constructor for {@link GVRSceneObject} for different combinations
+ * of {@link GVRMesh} or Future<{@link GVRMesh}>, {@link GVRTexture} or Future<
+ * {@link GVRTexture}>.
+ * 
+ * @param <V>
+ *            Internal type to be wrapped. It can be either mesh or texture.
+ */
+public class FutureWrapper<V> implements Future<V> {
+
+    private V value_ = null;
```

Comment:
Fwiw, I would code this as a `final` field, but I don't think that you need to change this.